### PR TITLE
Fix reordering in parallel

### LINKED ISSUE #86: Reordering test fails when run in parallel
Current test is performed in serial, which passes. However if ran in parallel it fails with SIGFPE, with @Nanoseb tracked this down to it should be passed `n_padded` rather than `n_loc`.

Proposal:
1) Duplicate test for serial and parallel execution
2) Rewrite ordering operations to us padded values

### PR DESCRIPTION
**Pull request by Nanoseb**
This PR fixes parallel reordering

Closes #86

### CODE REVIEW

File: src/omp/backend.f90 [hunk 1]
```diff
@@ -316,22 +316,22 @@ subroutine reorder_omp(self, u_, u, direction)
     select case (direction)
```
[the remark this answers is not in the document]
Comment (Nanoseb) [reply src/omp/backend.f90:316]: This select statement will be removed once data is in the field itself. So I won't bother doing it here for now.

**Comment by semi-h** [comment]
It looks like there are a few lines fprettify will complain, I think this can be merged once the fprettify output is clear.

File: tests/test_reordering.f90 [hunk 2]
```diff
@@ -117,11 +117,11 @@ program test_reorder
...
       do i = 1, xdirps%n
         call test_index_reversing(pass_X, i, j, k, DIR_X, &
-                                  SZ, xdirps%n, ydirps%n, zdirps%n)
+                                  SZ, allocator%xdims_padded(2), allocator%ydims_padded(2), allocator%zdims_padded(2))
         call test_index_reversing(pass_Y, i, j, k, DIR_Y, &
-                                  SZ, xdirps%n, ydirps%n, zdirps%n)
+                                  SZ, allocator%xdims_padded(2), allocator%ydims_padded(2), allocator%zdims_padded(2))
         call test_index_reversing(pass_Z, i, j, k, DIR_Z, &
-                                  SZ, xdirps%n, ydirps%n, zdirps%n)
+                                  SZ, allocator%xdims_padded(2), allocator%ydims_padded(2), allocator%zdims_padded(2))
```
Comment (pbartholomew08) [inline tests/test_reordering.f90:124]: Ah - this is where I went wrong, I assumed dims(1) == x, dims(2) == y, etc.

File: src/omp/backend.f90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (pbartholomew08) [reply src/omp/backend.f90:316]: Is `ndir_groups` still a good name (even if temporary)?

**Review by pbartholomew08 — APPROVED** [review]

File: src/omp/backend.f90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nanoseb) [reply src/omp/backend.f90:316]: I believe so. It is the number of groups in the `dir` direction. Is other location of the code it is called `nx_groups`, so a generic version of that name is `ndir_groups`. But if the `groups` terminology isn't clear maybe that's something to change though.

**Pull request merged**